api/proto/config/storage.proto

```diff
@@ -193,3 +193,41 @@ message Volume {
   string displayName = 7;  // Optional friendly name echo'ed in info message
   bool clear_text = 8;  // Flag to indicate the volume encryption needed or not
 }
+
+// DiskConfigType is the desired configuration of disks
+enum DiskConfigType {
+  DISK_CONFIG_TYPE_UNSPECIFIED = 0; // no configured, do nothing
+  DISK_CONFIG_TYPE_EVEOS = 1; // the disk EVE is installed on
+  DISK_CONFIG_TYPE_PERSIST = 2; // the disk is separate persist partition, not zfs
+  DISK_CONFIG_TYPE_ZFS_ONLINE = 3; // included in zfs and online
+  DISK_CONFIG_TYPE_ZFS_OFFLINE = 4; // included in zfs and offline
+  DISK_CONFIG_TYPE_APPDIRECT = 5; // for direct assignment
+  DISK_CONFIG_TYPE_UNUSED = 6; // removed from zfs/app-direct
+}
+
+//DiskDescription describes disk
+//we can use different data to locate disk in the system
+message DiskDescription {
+  string name = 1; // bus-related name, for example: /dev/sdc
+  string logical_name = 2; // logical name, for example: disk3
+  string serial = 3; // serial number of disk
+}
+
```

Comment:
Can we put this in evecommon directory so that it can be shared between config and info subdirs e.g., for https://github.com/lf-edge/eve/pull/2529?

Reply:
Done